backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -162,7 +164,11 @@ public EndpointPage getEndpoints(
         for (Endpoint endpoint: endpoints) {
             // Fetch the secrets from Sources.
             if (this.featureFlipper.isSourcesUsedAsSecretsBackend()) {
-                this.secretUtils.loadSecretsForEndpoint(endpoint);
+                try {
+                    this.secretUtils.loadSecretsForEndpoint(endpoint);
+                } catch (final SourcesException e) {
+                    Log.error(e.getMessage());
```

Comment:
True! I have modified it to be `Log.error(e.getMessage(), e)`, since the error message is already prepared to be logged, with all the details we could require. Thanks!